Meadowlark-js/packages/meadowlark-core/src/middleware/ValidateDocumentMiddleware.ts

```diff
@@ -6,20 +6,23 @@
 import { validateDocument } from '../validation/DocumentValidator';
 import { writeDebugObject, writeRequestToLog } from '../Logger';
 import { MiddlewareModel } from './MiddlewareModel';
-import { FrontendRequest } from '../handler/FrontendRequest';
 
 const moduleName = 'core.middleware.ValidateDocumentMiddleware';
 
 /**
  * Validates JSON document shape
  */
-export async function documentValidation({ frontendRequest, frontendResponse }: MiddlewareModel): Promise<MiddlewareModel> {
+async function documentValidation(
+  { frontendRequest, frontendResponse }: MiddlewareModel,
+  isUpdate: boolean = false,
```

Comment:
frontendRequest.action tells you if it's an update or insert. You can probably shrink this back down to a single middleware function.